src/lib/y2users/users_simple/reader.rb

```diff
@@ -99,7 +120,9 @@ def attr_value(attr, user)
       # @param user [Hash] a user representation in the format used by UsersSimple
       # @return [Password]
       def create_password(user)
-        new_password(user["userPassword"], user["encrypted"]).tap do |password|
+        create_method = user["encrypted"] ? :create_encrypted : :create_plain
+
+        Password.send(create_method, user["userPassword"]).tap do |password|
```

Comment:
at least use `public_send` other way could be to use passwordValue class and here just call `klass.new(user["userPassword"])`